src/sdk/pynni/nni/compression/torch/pruners.py

```diff
@@ -70,14 +72,16 @@ class AGP_Pruner(Pruner):
     https://arxiv.org/pdf/1710.01878.pdf
     """
 
-    def __init__(self, model, config_list, optimizer):
+    def __init__(self, model, config_list, optimizer=None):
```

Comment:
optimizer cannot be None in AGP pruner because it is an iterative pruner

Reply:
or give default value None to the argument optimizer, and check optimizer in `__init__`, if it is None raise error with message "AGP pruner is an iterative pruner, please pass optimizer of the model to it"

Reply:
Thanks, fixed.